File: tax_excel/tax_excel/report/payroll_tax_export/payroll_tax_export.py
# Copyright (c) 2022, Jide Olayinka and contributors
# For license information, please see license.txt

import frappe

def execute(filters=None):
	"""Tax report view"""
	condition_date = ""
	condition_pm = ""
	if filters.date_from_filter and filters.date_to_filter :
		if filters.date_from_filter == None:
			filters.date_from_filter = frappe.datetime.get_today()
		if filters.date_to_filter == None:
			filters.date_to_filter = frappe.datetime.get_today()
		condition_date = "AND start_date BETWEEN '"+ filters.date_from_filter + \
        "' AND '" + filters.date_to_filter + "'"

	columns = [
    {'fieldname':'name','label':'Salary Slip ID','width':'200'},
    {'fieldname':'employee','label':'Employee','width':'120'},
	{'fieldname':'employee_name','label':'Employee Name','width':'150'},
	{'fieldname':'date_of_joining','label':'Date Join','width':'100'},
	{'fieldname':'branch','label':'Current State','width':'100'},    
	{'fieldname':'designation','label':'Designation','width':'100'},
	{'fieldname':'department','label':'Department','width':'150'},
	{'fieldname':'contrib','label':'Tax','width':'150','fieldtype': 'Float'},
	{'fieldname':'liabil','label':'Overtime Tax','width':'150','fieldtype': 'Float'},
	]
	
	nw_data = "SELECT name, employee,employee_name,date_of_joining,department,designation,grade,start_date,end_date,contrib ,liabil ,branch FROM (select \
		s.name,s.employee,s.employee_name,s.start_date,s.end_date,s.docstatus, \
		e.tax_id,e.date_of_joining,e.department,e.current_state_of_abode_ as branch,e.designation,e.grade, e.pension_id,e.name_of_pension_manager,e.pension_manager,e.employee_name as femployee, \
		v.contrib , v.liabil ,v.parent \
 		from `tabSalary Slip` s left join `tabEmployee` e on s.employee = e.name \
		LEFT JOIN \
				(SELECT Distinct k.parent, \
        					IFNULL((select d.amount from `tabSalary Detail` d where d.parentfield='deductions'and d.salary_component ='Tax' and d.parent=k.parent),0) as contrib, \
        					IFNULL((select d.amount from `tabSalary Detail` d where d.parentfield='deductions'and d.salary_component = 'Overtime Tax'and d.parent=k.parent),0) as liabil \
        FROM (select d.amount,d.parent,d.salary_component from `tabSalary Detail` d where d.salary_component in ('Tax','Overtime Tax') ) k \
			) v ON s.name = v.parent ) a WHERE branch <> '' {} ".format(condition_date) 

	data = frappe.db.sql(nw_data, as_dict=1,)

	return columns, data

File: tax_excel/tax_excel/report/payroll_tax_export/payroll_tax_export.js
// Copyright (c) 2022, Jide Olayinka and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Payroll Tax Export"] = {
	"filters": [
		{
			fieldname: "date_from_filter",
			label: "Time From Filter",
			fieldtype: "Date",

		},
		{
			fieldname: "date_to_filter",
			label: "Time To Filter",
			fieldtype: "Date",

		}

	],
	onload: function(report){
		report.page.add_inner_button(__("Export To Excel"), function(){
			let fm_d = frappe.query_report.get_filter_value('date_from_filter');
			let to_d = frappe.query_report.get_filter_value('date_to_filter');

			if (fm_d != undefined && fm_d != "" && to_d != undefined && to_d != "") {
				//method: "tax_excel.tax_excel.utils.pay_roll_tax_report",
				const args = {
					cmd: 'tax_excel.tax_excel.utils.pay_roll_tax_report',
					report_name: `${Date.now()}.xlsx`,
					from_date:fm_d,
					to_date:to_d,
				};
				open_url_post(frappe.request.url, args);
			}
			else{
				//
				frappe.msgprint("Please select a valid 'from date' and or 'to date'");
			}
			
		});
		report.page.change_inner_button_type('Export To Excel',null, 'success');
	},
	/*"formatter": function (value, row, column, data, default_formatter) {
		value = default_formatter(value, row, column, data);

		if (column.fieldname == "contrib" && data && data.debit > 0) {
			value = "<span style='color:red'>" + value + "</span>";
		}
		else if (column.fieldname == "liabil" && data && data.credit > 0) {
			value = "<span style='color:green; font-weight:bolder'>" + value + "</span>";
		}

		return value;
	}*/
};
